### PR TITLE
mpv: Add extra_script hooks and add generic build hook.

### LINKED ISSUE #1200: Custom patches for mpv
How do I apply patches for mpv? I don't see any supported commands (do_autoreconf, do_configure etc.) in the whole mpv building process so I can't make mpv_extra.sh like this: 
```
_pre_configure(){
    do_patch "https://patch-diff.githubusercontent.com/raw/mpv-player/mpv/pull/5411.patch" am
}
```

### PR DESCRIPTION
**Pull request by 1480c1**
Closes #1200

### CODE REVIEW

File: build/media-suite_compile.sh [hunk 1]
```diff
@@ -1980,6 +1980,7 @@ if [[ $mpv != "n" ]] && pc_exists libavcodec libavformat libswscale libavfilter;
             done
         fi
 
+        extra_script pre configure
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: Haven't thought of this until now, but do the functions have access to the variables in this scope? i.e. can they add/remove stuff to `mpv_cflags` or `mpv_ldflags`?

Comment (1480c1) [reply build/media-suite_compile.sh]: Yea, it seems they do

Comment (1480c1) [reply build/media-suite_compile.sh]: they should have access to all of the variables within that sections

Comment (wiiaboo) [reply build/media-suite_compile.sh]: Cool.

**Comment by wiiaboo** [comment]
Good work on this.

**Comment by 1480c1** [comment]
Thanks

**Pull request merged**